### DISCUSSION TITLE
Understanding the Background Execution Limits of Remix  Loader Functions

### DISCUSSION THREAD

**Discussion by maxman46** [remix-run/remix#8033] [Q&A]
I've recently debugged a route, and it was supposed to finish executing the loader function within two minutes. However, even though the view was successfully returned to the browser, the loader function continued running in the background. I'm wondering if there's a time limit on how long it can run like this in the background.

If there isn't a time limit or if the limit is quite generous, could Remix routes potentially be utilized to run extended background tasks in loader functions simply by triggering them through URLs?

**Comment by sergiodxa** [comment]
I don't think Remix itself keeps the loader running, maybe only for `defer` responses where a promise is not resolved yet.

The browser typically will timeout the request eventually if the response never closes.
